File: 백준/Bronze/4597. 패리티/패리티.py
while(True):
  bit = input()
  if(bit == "#"):
    break

  num = 0
  for i in range(len(bit) - 1):
    if bit[i] == '1':
      num += 1
    print(bit[i], end = '')
  
  if(bit[len(bit) - 1] == 'e'):
    if num % 2 == 0:
      print('0')
    else:
      print('1')
  else:
    if num % 2 == 0:
      print('1')
    else:
      print('0')

File: 백준/Bronze/4597. 패리티/패리티.c
#include <stdio.h>
#include <string.h>

int main(void) { 
   char bit[33];

   while(1){
      int num = 0;
      scanf("%s", bit);
      if(strcmp(bit, "#") == 0){
         break;
      }

      int length = strlen(bit);

      for(int i = 0; i < length - 1; i++){
         if(bit[i] == '1'){
            num += 1;
         }
         printf("%c", bit[i]);
      }

      if(bit[length - 1] == 'e'){
         if(num % 2 == 0){
            printf("0");
         }
         else{
            printf("1");
         }
      }
      else{
         if(num % 2 == 0){
            printf("1");
         }
         else{
            printf("0");
         }
      }
      printf("\n");
   }
   
   return 0;
}